src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_OF_PAY = 10;
+    private static final int HOURS_WORKED = 2;
+    private static final int SALARY_PER_HOUR = 3;
+    private static final int START_OF_RANGE = 0;
+    private static final String SPACE = " ";
+    private static final String REPORT_MESSAGE = "Report for period ";
+    private static final String SEPARATE = " - ";
+    private StringBuilder sb = new StringBuilder();
+    private int salary = 0;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateStart = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate dateLast = LocalDate.parse(dateTo, FORMATTER);
+        String[] resultToReport = new String[names.length];
+
+        for (int i = 0; i < names.length; i++) {
+            salary = 0;
+            for (String dataArr : data) {
+                if (dataArr.contains(names[i])) {
+                    LocalDate currentDate = LocalDate
+                            .parse(dataArr.substring(START_OF_RANGE, DATE_OF_PAY), FORMATTER);
+                    sb.delete(START_OF_RANGE, sb.length());
```

Comment:
I think that after we move `StringBuilder` to the method, we can remove line with `sb.delete` call

Reply:
> I think that after we move `StringBuilder` to the method, we can remove line with `sb.delete` call

You mean creating new Stringbuilder in the loop instead of cleaning like i did? Sorry not sure if you meant that or something else.

Reply:
not in the loop, at the beginning of the method, as you did in your next commit